### PR TITLE
[DOCS] Clarify copy_to behavior with strict dynamic mappings

### LINKED ISSUE #97817: copy_to description does not mention usage of path
### Elasticsearch Version

8.8

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Docker

### Problem Description

I have recently encountered an issue regarding the copy_to field in Elasticsearch mappings, specifically when combined with dynamic mappings set to strict.

According to the current Elasticsearch documentation, the copy_to parameter allows the value of one field to be copied into another field automatically during indexing. However, it seems that when the full path is not specified in the copy_to directive, it points to the root of the field structure rather than the nested field, which may not be intuitive to users.

In cases where dynamic mapping is set to strict, this behavior can lead to a strict_dynamic_mapping_exception if a document is indexed with a field that has not been explicitly defined in the mapping. This happens even if the field is present as a target in a copy_to directive, but the full path is not specified.

I suggest that the documentation be updated to clarify this point. It would be beneficial to highlight the importance of using the full path in copy_to directives, particularly when dynamic mapping is set to strict. Additionally, the potential exception that could be thrown due to the non-fully qualified path in copy_to under strict dynamic mapping should also be mentioned.

This clarification could prevent future misunderstandings and exceptions for Elasticsearch users.

### Steps to Reproduce

1. Create an index with dynamic mapping set to `strict` and a `copy_to` directive without a fully qualified path:

```json
PUT /test_index
{
  "mappings": {
    "dynamic": "strict",
    "properties": {
      "description": {
        "properties": {
          "notes": {
            "type": "text",
            "copy_to": [
              "notes_raw"
            ],
            "analyzer": "standard",
            "search_analyzer": "standard"
          },
          "notes_raw": {
            "type": "keyword"
          }
        }
      }
    }
  }
}
```

2. Attempt to index a document:

```json
PUT /test_index/_doc/1
{
  "description": {
    "notes": "This is a test note"
  }
}
```

3. Observe the resulting `strict_dynamic_mapping_exception`, stating that dynamic introduction of `notes_raw` within `[_doc]` is not allowed.

This error occurs because Elasticsearch interprets the `copy_to` target of `notes_raw` as a root-level field, not a nested field under `description`. However, since `notes_raw` is not explicitly defined in the root of the mapping but under the `description` field and dynamic mapping is set to `strict`, indexing fails with the mentioned exception.

The solution would be to use a fully qualified path in the `copy_to` directive, like so: `"copy_to": [ "description.notes_raw" ]`. But this is currently not clear in the documentation, leading to potential confusion.

Please consider updating the documentation to highlight this behavior.

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by leemthompo**
Closes https://github.com/elastic/elasticsearch/issues/97817

- Group dynamic mapping behavior into its own section
- Mention how `strict` requires fully qualified paths when working with nested fields

### [URL preview](https://elasticsearch_bk_111408.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/copy-to.html)

### CODE REVIEW

File: docs/reference/mapping/params/copy-to.asciidoc [hunk 1]
```diff
@@ -68,12 +68,58 @@ Some important points:
 `field_1` to `field_2` and `copy_to` on `field_2` to `field_3` expecting 
 indexing into `field_1` will eventuate in `field_3`, instead use copy_to 
 directly to multiple fields from the originating field.
```
Comment (shainaraskas) [inline docs/reference/mapping/params/copy-to.asciidoc:70]: while you're here, this sentence makes zero sense

File: docs/reference/mapping/params/copy-to.asciidoc [hunk 2]
```diff
@@ -68,12 +68,58 @@ Some important points:
...
+
+NOTE: `copy_to` is _not_ supported for field types where values take the form of objects, e.g. `date_range`
```
Comment (shainaraskas) [inline docs/reference/mapping/params/copy-to.asciidoc:72]: suggest skipping the italics

```suggestion
NOTE: `copy_to` is not supported for field types where values take the form of objects, e.g. `date_range`.
```

File: docs/reference/mapping/params/copy-to.asciidoc [hunk 3]
```diff
@@ -68,12 +68,58 @@ Some important points:
...
+
+[float]
+[[copy-to-dynamic-mapping]]
+==== Dynamic mapping
+
+Here are some important points to consider when using `copy_to` with dynamic mappings:
```
Comment (shainaraskas) [inline docs/reference/mapping/params/copy-to.asciidoc:78]:
```suggestion
Consider the following points when using `copy_to` with dynamic mappings:
```

File: docs/reference/mapping/params/copy-to.asciidoc [hunk 4]
```diff
@@ -68,12 +68,58 @@ Some important points:
...
-dynamically added to the index mappings. If `dynamic` is set to `false`, the
+dynamically added to the index mappings.
+* If `dynamic` is set to `false`, the
 target field will not be added to the index mappings, and the value will not be
-copied. If `dynamic` is set to `strict`, copying to a non-existent field will
+copied.
+* If `dynamic` is set to `strict`, copying to a non-existent field will
 result in an error.
++
+** `copy_to` fields must specify the full path if the target field is _nested_.
```
Comment (shainaraskas) [inline docs/reference/mapping/params/copy-to.asciidoc:90]: again, preference is no italics. would also swap this around to make people spot `nested` first

```suggestion
**  If the target field is nested, then `copy_to` fields must specify the full path to the nested field.
```

**Review by shainaraskas — APPROVED** [review]
approved w/ some minor suggestions

File: docs/reference/mapping/params/copy-to.asciidoc (hunk 1 above)
Comment (leemthompo) [reply docs/reference/mapping/params/copy-to.asciidoc:70]: rejigged

**Pull request merged**